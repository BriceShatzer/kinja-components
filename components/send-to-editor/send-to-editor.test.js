import * as React from 'react';

import { shallow } from 'enzyme';

import SendToEditor from './send-to-editor';

const noop = jest.fn();

const defaultValues = { notes: '', status: 'Freelancer Filed' };
const stubElement = ({
	onSubmit = noop,
	onCancel = noop,
	initialValues = defaultValues
} = {}) => (
	<SendToEditor
		language="en-US"
		onSubmit={onSubmit}
		onCancel={onCancel}
		initialValues={initialValues}
	/>
);

describe('<SendToEditor />', () => {
	it('should call onCancel when cancel', () => {
		const cancelHandler = jest.fn();
		const wrapper = shallow(stubElement({
			onCancel: cancelHandler
		}));

		wrapper.find({ weight: 'secondary' }).at(0).simulate('click');
		expect(cancelHandler).toHaveBeenCalledTimes(1);
	});

	it('should call onSubmit when submit', () => {
		const submitHandler = jest.fn();
		const wrapper = shallow(stubElement({
			onSubmit: submitHandler
		}));

		wrapper.find({ weight: 'primary' }).at(0).simulate('click');
		expect(submitHandler).toHaveBeenCalledTimes(1);
	});

	it('should render component with default state "Freelancer Filed" selected', () => {
		expect(shallow(stubElement())).toMatchSnapshot();
	});

	it('should set notes from initialValues', () => {
		const changedState = stubElement({
			initialValues: { notes: 'sic transit gloria mundi' }
		});

		expect(changedState).toMatchSnapshot();
	});

	it('should set status after changing the ButtonGroup', () => {
		const onSubmit = jest.fn();
		const wrapper = shallow(stubElement({ onSubmit }));

		// Change status
		wrapper.find('ButtonGroup').simulate('change', 'Ready for Copy Desk');
		// Submit
		wrapper.find({ weight: 'primary' }).at(0).simulate('click');
		expect(onSubmit.mock.calls[0]).toMatchSnapshot();
	});

	it('should set notes after changing notes', () => {
		const onSubmit = jest.fn();
		const wrapper = shallow(stubElement({ onSubmit }));

		// Change notes
		wrapper.find('Textarea').simulate('change', 'muhaha');
		// Submit
		wrapper.find({ weight: 'primary' }).at(0).simulate('click');
		expect(onSubmit.mock.calls[0]).toMatchSnapshot();
	});
});
