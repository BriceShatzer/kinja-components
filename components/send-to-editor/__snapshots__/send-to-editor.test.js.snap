// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SendToEditor /> should render component with default state "Freelancer Filed" selected 1`] = `
<div
  className="send-to-editor-form"
>
  <h2
    className="modal-title"
  >
    Send to Editors
  </h2>
  <div
    className="form-group flex-row flex-row--align-middle"
  >
    <div
      className="flex-row__column small-4"
    >
      <h3
        className="form-group_label"
      >
        Article Status
      </h3>
    </div>
    <div
      className="flex-row__column small-8"
    >
      <ButtonGroup
        onChange={[Function]}
      >
        <ButtonGroupItem
          label="Freelancer Filed"
          selected={true}
          value="Freelancer Filed"
        />
        <ButtonGroupItem
          label="Ready for Copy Desk"
          value="Ready for Copy Desk"
        />
      </ButtonGroup>
    </div>
  </div>
  <div
    className="form-group flex-row"
  >
    <div
      className="flex-row__column small-4"
    >
      <h3
        className="form-group_label"
      >
        Additional Notes
      </h3>
    </div>
    <div
      className="flex-row__column small-8"
    >
      <Textarea
        label="Run date, word count, graphics, photo credit, etc."
        name="notes"
        onChange={[Function]}
        rows={5}
        value=""
      />
    </div>
  </div>
  <footer
    className="form-buttons"
  >
    <Button
      className="form-button"
      label="Cancel"
      onClick={[MockFunction]}
      type="button"
      weight="secondary"
    />
    <Button
      className="form-button"
      label="Send"
      onClick={[Function]}
      type="submit"
      weight="primary"
    />
  </footer>
</div>
`;

exports[`<SendToEditor /> should set notes after changing notes 1`] = `
Array [
  Object {
    "notes": "muhaha",
    "status": "Freelancer Filed",
  },
]
`;

exports[`<SendToEditor /> should set notes from initialValues 1`] = `
<SendToEditorForm
  initialValues={
    Object {
      "notes": "sic transit gloria mundi",
    }
  }
  language="en-US"
  onCancel={[MockFunction]}
  onSubmit={[MockFunction]}
/>
`;

exports[`<SendToEditor /> should set status after changing the ButtonGroup 1`] = `
Array [
  Object {
    "notes": "",
    "status": "Ready for Copy Desk",
  },
]
`;
